system_metrics_collector/src/topic_statistics_collector/subscriber_topic_statistics.cpp

```diff
@@ -0,0 +1,271 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```

Comment:
Implementation is in a separate file because personally I think it looks cleaner and more readable that way. However, I will defer to the [coding standard](https://google.github.io/styleguide/cppguide.html#Self_contained_Headers) and place them in one file going forward.